views/js/controller/Delivery/monitoring.js

```diff
@@ -452,6 +448,78 @@ define([
                 }
 
                 /**
+                 * Create a warning message for execBulkAction()
+                 * @param {String} action
+                 * @param {String[]} selection
+                 * @param {Object[]} deniedResources
+                 */
+                function buildWarningMessage(action, selection, deniedResources) {
+                    var maxReasons = 2;
+                    var warningAction;
+                    var warningReason;
+                    var warningReasonOmission = __('...');
+
+                    deniedResources = deniedResources || [];
+                    selection = selection || [];
+
+                    warningAction = getWarningAction(action, deniedResources.length > 1 || selection.length > 1);
+                    warningReason = _(deniedResources || [{}])
+                        .slice(0, maxReasons)
+                        .map(function (deniedResource, index) {
+                            return getWarningReason(deniedResource.reason, deniedResource.id || (index + 1));
+                        })
+                        .join(' ');
+
+                    return __('%s %s%s', warningAction, warningReason, (deniedResources.length > maxReasons ? warningReasonOmission : ''));
+
+                    function getWarningAction(actionName, isPlural) {
+                        if (isPlural) {
+                            return {
+                                authorize: __('Cannot authorize test sessions.'),
+                                pause: __('Cannot pause test sessions.'),
+                                print: __('Cannot print test sessions.'),
+                                report: __('Cannot generate report on test sessions.'),
+                                terminate: __('Cannot terminate test sessions.'),
+                                time: __('Cannot extend test sessions.'),
+                                default: __('No report available for test sessions.')
+                            }[actionName || 'default'];
+                        } else {
+                            return {
+                                authorize: __('Cannot authorize test session.'),
+                                pause: __('Cannot pause test session.'),
+                                print: __('Cannot print test session.'),
+                                report: __('Cannot generate report on test session.'),
+                                terminate: __('Cannot terminate test session.'),
+                                time: __('Cannot extend test session.'),
+                                default: __('No report available for test session.')
+                            }[actionName || 'default'];
+                        }
+                    }
+
+                    function getWarningReason(reason, testId) {
```

Comment:
Those two function are nice, but maybe should they be placed into a proper helper, like `taoProctoring/helper/status`, should not be ?